utils/monitoring_tools/pytorch_model_internals/scripts/torch_watcher_example.py

```diff
@@ -0,0 +1,144 @@
+    """Training function demonstrating different ways to use TorchWatcher."""
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+    criterion = nn.MSELoss()
+    n_batches = len(X_train) // batch_size
+
+    for epoch in range(n_epochs):
+        model.train()
+        train_loss = 0.0
+
+        # Training batches
+        for i in range(n_batches):
+            start_idx = i * batch_size
+            end_idx = start_idx + batch_size
+
+            x_batch = X_train[start_idx:end_idx]
+            y_batch = y_train[start_idx:end_idx]
+
+            # Forward pass
+            output = model(x_batch)
+            loss = criterion(output, y_batch)
+
+            # Backward pass
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+
+            # Track metrics during training
+            if epoch < 5:  # First 5 epochs: track everything
+                watcher.watch(step=epoch * n_batches + i, namespace="train")
+            else:  # Later epochs: track only gradients for efficiency
+                watcher.watch(
+                    step=epoch * n_batches + i,
+                    track_activations=False,
+                    track_parameters=False,
+                    track_gradients=True,
+                    namespace="train",
+                )
+
+            train_loss += loss.item()
+
+        # Average training loss
+        train_loss /= n_batches
+
+        # Validation
+        model.eval()
+        with torch.no_grad():
+            val_output = model(X_val)
+            val_loss = criterion(val_output, y_val)
+
+            # Track metrics during validation
+            watcher.watch(
+                step=epoch,
+                track_activations=True,
+                track_gradients=False,
+                track_parameters=False,
+                namespace="validation",
+            )
+
+        # Log metrics
+        watcher.run.log_metrics(
+            data={"train/loss": train_loss, "val/loss": val_loss.item()}, step=epoch
+        )
+
+        if (epoch + 1) % 10 == 0:
+            print(
+                f"Epoch [{epoch+1}/{n_epochs}], "
+                f"Train Loss: {train_loss:.4f}, "
+                f"Val Loss: {val_loss.item():.4f}"
+            )
+
+
+def main():
+    # Initialize Neptune run
+    run = Run(
+        experiment_name="torch-watcher-example",
+    )
+
+    # Generate data
+    X_train, y_train = generate_data(n_samples=1000)
+    X_val, y_val = generate_data(n_samples=200)
+
+    # Create model and watcher
+    model = SimpleNet()
+
+    # Initialize watcher with specific layer types to track
+    watcher = TorchWatcher(
+        model,
+        run,
+        track_layers=[nn.Linear, nn.ReLU],  # Track only Linear and ReLU layers
+        tensor_stats=["mean", "norm"],  # Track mean and norm statistics
+        base_namespace="model_metrics",  # Default namespace for all metrics
```

Comment:
`model_metrics` can also include evaluation metrics like `acc`, `loss`, etc.

```suggestion
        base_namespace="model_internals",  # Default namespace for all metrics
```